### PR TITLE
Refactor main to follow updated best practices

### PR DESCRIPTION
**Pull request by phillebaba**
This change tries to bring code in line with Xenit best practices.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,6 @@ go 1.18
```
Comment (simongottschlag) [inline go.mod:4]: Looks like 1.18 is used, maybe worth bumping to 1.19

File: main.go [hunk 2]
```diff
@@ -7,134 +7,98 @@ import (
...
-}
-
 func main() {
+	var args struct {
+		Addr           string `arg:"--addr,required" default:":8080"`
+		MetricsAddr    string `arg:"--metrics-addr,required" default:":9090"`
+		CfgPath        string `arg:"--config,required"`
+		KubeconfigPath string `arg:"--kubeconfig"`
+	}
+	arg.MustParse(&args)
```
Comment (simongottschlag) [inline main.go:35]: Should we use something that panics?

I usually separate it out to a newConfig() like this:
```go
func newConfig(args []string) (Config, error) {
	cfg := Config{}

	parser, err := arg.NewParser(arg.Config{
		Program:   "azcagit",
		IgnoreEnv: false,
	}, &cfg)
	if err != nil {
		return Config{}, err
	}

	err = parser.Parse(args)
	if err != nil {
		return Config{}, err
	}

	return cfg, nil
}
```

And then in main:
```go
	cfg, err := config.NewConfig(os.Args[1:])
	if err != nil {
		log.Error(err, "unable to load config")
		os.Exit(1)
	}
```

Comment (phillebaba) [reply main.go:35]: I personally think this is fine beacuse the function does a bunch of other usefult stuff like printing helper info to stdout.

Comment (phillebaba) [reply main.go:35]: I will have another look and see if there is a better way of doing this.

Comment (ahockersten) [reply main.go:35]: @simongottschlag isn't logging an error and then calling `os.Exit()` basically functionally equivalent to panicking though? The end result of both designs is that your application dies on startup when config is missing.

**Review by ahockersten — APPROVED** [review]

File: main.go (hunk 2 above)
Comment (simongottschlag) [reply main.go:35]: https://go-proverbs.github.io/

> Don't panic

Comment (phillebaba) [reply main.go:35]: I checked the code now and dont really see where the code would panic? It does an `os.Exit(1)`

**Pull request merged**